processing/src/main/java/io/druid/segment/LongDimensionIndexer.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.segment;
+
+import com.google.common.collect.Lists;
+import io.druid.collections.bitmap.BitmapFactory;
+import io.druid.collections.bitmap.MutableBitmap;
+import io.druid.query.dimension.DimensionSpec;
+import io.druid.segment.data.Indexed;
+import io.druid.segment.incremental.IncrementalIndex;
+import io.druid.segment.incremental.IncrementalIndexStorageAdapter;
+
+import java.util.List;
+
+public class LongDimensionIndexer implements DimensionIndexer<Long, Long, Long>
+{
+  @Override
+  public Long processRowValsToUnsortedEncodedKeyComponent(Object dimValues)
+  {
+    if (dimValues instanceof List) {
+      throw new UnsupportedOperationException("Numeric columns do not support multivalue rows.");
+    }
+
+    return DimensionHandlerUtils.convertObjectToLong(dimValues);
+  }
+
+  @Override
+  public Long getSortedEncodedValueFromUnsorted(Long unsortedIntermediateValue)
+  {
+    return unsortedIntermediateValue;
+  }
+
+  @Override
+  public Long getUnsortedEncodedValueFromSorted(Long sortedIntermediateValue)
+  {
+    return sortedIntermediateValue;
+  }
+
+  @Override
+  public Indexed<Long> getSortedIndexedValues()
+  {
+    return null;
```

Comment:
Changed this and the bitmap-related method to throw UnsupportedOperationException